### DISCUSSION TITLE
Maximum execution time of 30 seconds exceeded

### DISCUSSION THREAD

**Discussion by ashikck** [opcodesio/log-viewer#216] [Q&A] [answered]
Maximum execution time of 30 seconds exceeded at /var/www/html/letsbid/vendor/opcodesio/log-viewer/src/LogIndexChunk.php:57

**Comment by arukompas** [comment]
Hey @ashikck , how big is the log file that you're trying to open?

**Reply by ashikck** [reply]
Laravel log file is almost a GB and i have deleted log and made logging daily. now its working fine. Thank you

**Reply by arukompas** [reply]
@ashikck it should have no problems reading a gigabyte of logs, unless the PC or PHP configuration is slow.

Here's an example here locally, 2+ GB log file was processed (cached) in ~8 seconds:

<img width="1411" alt="Screenshot 2023-03-22 at 10 23 01" src="https://user-images.githubusercontent.com/8697942/226842507-0227fb68-89e5-441d-b873-bcb3094dfac2.png">

But I'm glad you found a solution! :)

**Reply by ashikck** [reply]
I have deployed in Digital Ocean VPC. Is there any server or PHP configuration or server  for loading large file like laravel log? Thank you

**Reply by arukompas** [reply, marked as answer]
Log Viewer doesn't consume a lot of memory. Even if the log file is 1 GB in size, it does not load all of it into memory.

You can see above that parsing a 2 GB log file only ever used maximum of 79 MB of memory. So it's definitely not about memory consumption.

I'm not familiar with the different Digital Ocean configurations, but if it's one of the cheaper ones then it's possible it's slower and it might take more than 30 seconds to process a large log file. I haven't tested it myself and I can't really help there, sorry :/
